### DISCUSSION TITLE
Adding my iCasa Zigbee 3.0 Keypad Pulse 8S to HomeBridge

### DISCUSSION THREAD

**Discussion by GroteGehaktBal** [Koenkk/zigbee2mqtt#11521] [Devices]
How can I add this iCasa Zigbee 3.0 Keypad Pulse 8S to homebridge. I'm getting an error in homebridge:

> [homebridge-z2m] Failed to setup stateless programmable switch for accessory 0x60a423fffe659823 from expose "{"access":1,"description":"Triggered action (e.g. a button click)","name":"action","property":"action","type":"enum","values":["on","recall_*","off","brightness_stop","brightness_move_up","brightness_move_down"]}", error: Error: Device found with a wildcard in the exposed possible values for the action, which cannot be mapped: recall_*

Model number: ICZB-KPD18S https://z2m.dev/devices/icasa/iczb-kpd18s.html
Homebridge plugin: homebridge-z2m v1.7.0
When pressing button S6 on the keypad this is the state in MQTT with it in legacy mode:

> {
    "brightness": 244,
    "rate": 50,
    "action": "recall_6",
    "action_group": 7937,
    "click": 6,
    "linkquality": 90
}

When legacy mode is set to false:

> {
    "brightness": 244,
    "linkquality": 102,
    "rate": 50,
    "action": "recall_6",
    "action_group": 7937
}

Could I use the following file somehow to fix this error?
https://github.com/koenkk/zigbee-herdsman-converters/blob/master/devices/icasa.js

Or can I somehow disable the "recall_*" actions and use the "click" instead?

Any help would be appreciated!

**Comment by itavero** [comment]
If you figure out which `recall_` values are being used, you can open up a PR on Koenkk/zigbee-herdsman-converters to put these in the `exposes` information of this device, instead of the wildcard value.

Looking at the picture of the device, I reckon it goes from `recall_1` up to `recall_6`.
